### PR TITLE
Invoke args2

### PR DESCRIPTION
**Pull request by adamabeshouse**
This one actually works!! But we'll need to update cbioportal-frontend typescript to latest to get the necessary features

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,6 @@
     "modulePathIgnorePatterns": [
       "/^((?!src).)/"
     ],
-    "testResultsProcessor": "<rootDir>/node_modules/ts-jest/coverageprocessor.js",
```
Comment (adamabeshouse) [inline package.json:59]: I deleted this because it was not making me not able to run `jest`. is it necessary to keep?

File: src/MobxPromise.ts [hunk 2]
```diff
@@ -44,11 +46,18 @@ export type MobxPromiseInputParams<R> = {
 	 */
 	onError?: (error:Error) => void,
 };
+
+export type AwaitType<InvokeType extends Function> =
+	InvokeType extends ((a0:infer A0)=>any) ? ()=>[MobxPromise<A0>] :
+		InvokeType extends ((a0:infer A0, a1:infer A1)=>any) ? ()=>[MobxPromise<A0>, MobxPromise<A1>] :
+		InvokeType extends ((a0:infer A0, a1:infer A1, a2:infer A2)=>any) ? ()=>[MobxPromise<A0>, MobxPromise<A1>, MobxPromise<A2>] :
```
Comment (adamabeshouse) [inline src/MobxPromise.ts:53]: todo: add support for more arguments

File: test/MobxPromise.spec.ts [hunk 3]
```diff
@@ -96,4 +96,19 @@ describe('MobxPromise', () => {
 			}, 200);
 		});
 	});
+
+	it('calls invoke with the arguments from await', (done)=>{
+		let mp1 = new MobxPromise(()=>new Promise<number>(resolve=>setTimeout(()=>resolve(5), 10)));
+		let mp2 = new MobxPromise(()=>new Promise<string>(resolve=>setTimeout(()=>resolve("hello"), 10)));
+		let mp3 = new MobxPromise({
+			await:()=>[mp1, mp2],
+			invoke:(a0:number, a1:string)=>{
```
Comment (adamabeshouse) [inline test/MobxPromise.spec.ts:105]: confirmed that typescript is actually doing inference here - if i change one of the argument types, TS fails; if i add an argument, TS fails

File: package.json (hunk 1 above)
Comment (adufilie) [reply package.json:59]: nope

**Pull request closed without merging**